polyfill/lib/duration.mjs

```diff
@@ -138,6 +138,12 @@ export class Duration {
   static from(...args) {
     return ES.CastDuration(...args);
   }
+  static fromNanoseconds(arg) {
+    return new this(0, 0, 0, 0, 0, 0, 0, 0, arg);
+  }
+  static fromMilliseconds(arg) {
+    return new this(0, 0, 0, 0, 0, 0, +arg, 0, 0);
```

Comment:
The `+` does ToNumber, but what happens then to any fractional part?

Regardless of what our own constructor does, we should probably do the type coercion before that, I think, because this becomes observable with subclasses

Reply:
The constructor would cast them ToInteger so that any fractional part would be lost. If we do anything here, we may want to use the fractional part as fractional seconds, which would mean this code is wrong.

Discuss!